omod/src/test/java/org/openmrs/module/idgen/web/controller/LogEntryControllerTest.java

```diff
@@ -0,0 +1,165 @@
+    }
+
+    @Test
+    public void shouldSearchAndReturnAListOfLogEntriesWithGivenSourceUuid() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        req.addParameter("source", LOG_ENTRY_SOURCE_UUID);
+        SimpleObject result = deserialize(handle(req));
+        Assert.assertNotNull(result);
+        Assert.assertEquals(3, Util.getResultsSize(result));
+    }
+
+    @Test
+    public void shouldSearchAndReturnNothingIfSourceUuidDoesNotMatch() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        req.addParameter("source", "invalid-source");
+        SimpleObject result = deserialize(handle(req));
+        Assert.assertEquals(0, Util.getResultsSize(result));
+    }
+
+    @Test
+    public void shouldSearchAndReturnAListOfLogEntriesWithIdentifier() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        req.addParameter("identifier", "H9");
+        SimpleObject result = deserialize(handle(req));
+        Assert.assertEquals(2, Util.getResultsSize(result));
+    }
+
+    @Test
+    public void shouldSearchAndReturnNothingIfIdentifierDoesNotMatch() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        req.addParameter("identifier", "invalid-identifier");
+        SimpleObject result = deserialize(handle(req));
+        Assert.assertEquals(0, Util.getResultsSize(result));
+
+    }
+
+    @Test
+    public void shouldSearchAndReturnAListOfLogEntriesGeneratedInDateRange() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        req.addParameter("fromDate", "2016-10-01 12:00:00.423");
+        req.addParameter("toDate", "2017-09-30 12:00:00.423");
+        SimpleObject result = deserialize(handle(req));
+        Assert.assertEquals(2, Util.getResultsSize(result));
+    }
+
+    @Test
+    public void shouldSearchAndReturnNothingIfNoLogsGeneratedWithinRange() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        req.addParameter("fromDate", "2015-10-01 12:00:00.423");
+        req.addParameter("toDate", "2016-09-30 12:00:00.423");
+        SimpleObject result = deserialize(handle(req));
+        Assert.assertEquals(0, Util.getResultsSize(result));
+    }
+
+    @Test
+    public void shouldSearchAndReturnAListOfLogEntriesWhoseCommentsContain() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        req.addParameter("comment", "New");
+        SimpleObject result = deserialize(handle(req));
+        Assert.assertEquals(2, Util.getResultsSize(result));
+    }
+
+    @Test
+    public void shouldSearchAndReturnNothingIfCommentsDontContain() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        req.addParameter("comment", "invalid comment");
```

Comment:
The test is missing.

Reply:
That was an oversight on my side. Thanks for pointing it out